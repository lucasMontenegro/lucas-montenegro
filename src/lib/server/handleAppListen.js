import globals from "lib/utils/globals"
import isProduction from "lib/utils/isProduction"
export default function handleAppListen (PORT) {
  return () => {
    const name = isProduction() ? `cluster worker ${globals.process.pid}` : `dev server`
    globals.console.log(`Node ${name}: listening on port ${PORT}`)
  }
}