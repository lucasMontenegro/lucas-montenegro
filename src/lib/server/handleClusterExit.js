import globals from "lib/utils/globals"
export default function handleClusterExit (worker, code, signal) {
  const { pid } = worker.process
  globals.console.log(`Node cluster worker ${pid} exited: code ${code}, signal ${signal}`)
}