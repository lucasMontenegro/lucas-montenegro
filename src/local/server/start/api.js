const express = require("express")
const todo = require("local/todo/router")
const api = module.exports = express.Router()
api.use(`/todo`, todo)