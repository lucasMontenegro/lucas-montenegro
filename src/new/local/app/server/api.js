const express = require("express")
const { default: todo } = require("new/local/todo/router")
const api = express.Router()
api.use(`/todo`, todo)
exports.default = api