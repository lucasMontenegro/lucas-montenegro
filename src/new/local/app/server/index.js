const cluster = require("cluster")
const os = require("os")
const { default: startHttp } = require("new/local/app/server/startHttp")
const { default: handleClusterExit } = require("new/local/app/server/handleClusterExit")
const { default: globals } = require("new/local/utils/globals")
exports.default = function start () {
  // Multi-process to utilize all CPU cores.
  if (globals.process.env.NODE_ENV === `production` && cluster.isMaster) {
    globals.console.log(`Node cluster master ${globals.process.pid} is running`)
    // Fork workers.
    for (let i = 0, n = os.cpus().length; i < n; i++) {
      cluster.fork()
    }
    cluster.on(`exit`, handleClusterExit)
  } else {
    startHttp()
  }
}