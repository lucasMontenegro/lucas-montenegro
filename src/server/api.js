import express from "express"
import todo from "lib/todo/router"
const api = express.Router()
api.use(`/todo`, todo)
export default api