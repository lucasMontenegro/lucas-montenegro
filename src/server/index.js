import startCluster from "lib/node/startCluster"
import httpServer from "lib/node/httpServer"
startCluster(() => httpServer())