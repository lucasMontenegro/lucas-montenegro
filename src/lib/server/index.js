import cluster from "cluster"
import os from "os"
import startHttp from "lib/server/startHttp"
import handleClusterExit from "lib/server/handleClusterExit"
import globals from "lib/utils/globals"
import isProduction from "lib/utils/isProduction"
export default function server (api) {
  // Multi-process to utilize all CPU cores.
  if (isProduction() && cluster.isMaster) {
    globals.console.log(`Node cluster master ${globals.process.pid} is running`)
    // Fork workers.
    for (let i = 0, n = os.cpus().length; i < n; i++) {
      cluster.fork()
    }
    cluster.on(`exit`, handleClusterExit)
  } else {
    startHttp(api)
  }
}