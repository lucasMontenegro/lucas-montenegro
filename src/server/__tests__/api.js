import todo from "lib/todo/router"
jest.mock(`lib/todo/router`, () => ({ __esModule: true, default: () => {} }))
describe(`server/api`, () => {
  it(`should use the right dependency versions`, () => {
    expect(jestUtils.getDependencies([`express`])).toMatchSnapshot()
  })
  it(`should run`, () => {
    require("server/api")
  })
})