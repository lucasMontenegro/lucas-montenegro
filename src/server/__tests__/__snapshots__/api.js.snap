// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server/api should use the right dependency versions 1`] = `
Object {
  "express": "^4.17.1",
}
`;
